examples/mnist/train_mnist.py

```diff
@@ -1,144 +1,100 @@
-                function_style = {'shape': 'record', 'fillcolor': '#6495ED',
-                                  'style': 'filled'}
-                g = computational_graph.build_computational_graph(
-                    (model.loss, ),
-                    variable_style=variable_style,
-                    function_style=function_style)
-                o.write(g.dump())
-            print('graph generated')
-
-        sum_loss += float(model.loss.data) * len(t.data)
-        sum_accuracy += float(model.accuracy.data) * len(t.data)
-    end = time.time()
-    elapsed_time = end - start
-    throughput = N / elapsed_time
-    print('train mean loss={}, accuracy={}, throughput={} images/sec'.format(
-        sum_loss / N, sum_accuracy / N, throughput))
-
-    # evaluation
-    sum_accuracy = 0
-    sum_loss = 0
-    for i in six.moves.range(0, N_test, batchsize):
-        x = chainer.Variable(xp.asarray(x_test[i:i + batchsize]),
-                             volatile='on')
-        t = chainer.Variable(xp.asarray(y_test[i:i + batchsize]),
-                             volatile='on')
-        loss = model(x, t)
-        sum_loss += float(loss.data) * len(t.data)
-        sum_accuracy += float(model.accuracy.data) * len(t.data)
-
-    print('test  mean loss={}, accuracy={}'.format(
-        sum_loss / N_test, sum_accuracy / N_test))
-
-# Save the model and the optimizer
-print('save the model')
-serializers.save_npz('mlp.model', model)
-print('save the optimizer')
-serializers.save_npz('mlp.state', optimizer)
+        chainer.cuda.get_device(args.gpu).use()  # Make a specified GPU current
+        model.to_gpu()  # Copy the model to the GPU
+
+    # Setup an optimizer
+    optimizer = chainer.optimizers.Adam()
+    optimizer.setup(model)
+
+    # Load the MNIST dataset
+    train, test = chainer.datasets.get_mnist()
+
+    train_iter = chainer.iterators.ShuffledIterator(train, args.batchsize)
+    test_iter = chainer.iterators.SequentialIterator(test, args.batchsize,
+                                                     repeat=False)
+
+    # Set up a trainer
+    updater = training.StandardUpdater(train_iter, optimizer, device=args.gpu)
+    trainer = training.Trainer(updater, (args.epoch, 'epoch'), out=args.out)
+
+    # Evaluate the model with the test dataset for each epoch
+    trainer.extend(extensions.Evaluator(test_iter, model, device=args.gpu))
+
+    # Dump a computational graph from 'loss' variable at the first iteration
+    trainer.extend(extensions.dump_graph('main/loss'))
+
+    # Take a snapshot at each epoch
+    trainer.extend(extensions.snapshot())
+
+    # Write a log of evaluation statistics for each epoch
+    trainer.extend(extensions.LogReport())
+
+    # Print selected entries of the log to stdout
+    # Entries prefixed by 'validation' are computed by the Evaluator extension
+    trainer.extend(extensions.PrintReport(
+        ['epoch', 'main/loss', 'validation/main/loss',
```

Comment:
A new user might be confused how these entries are derived. For example, explain what 'main' refers to, that Classifier reports the loss and accuracy.